trade-finance-manager-ui/server/controllers/utilisation-reports/update-utilisation-report-status/index.ts

```diff
@@ -22,24 +25,26 @@ const FORM_BUTTON_VALUES = {
 export type UpdateUtilisationReportStatusRequestBody = {
   _csrf: string;
   'form-button': string;
-  [key: `set-status--reportId-${string}`]: 'on';
+  [key: `set-status--reportId-${string}-currentStatus-${UtilisationReportReconciliationStatus}`]: 'on';
 };
 
-const getReportIdsFromBody = (body: undefined | UpdateUtilisationReportStatusRequestBody): string[] => {
+const getReportIdsAndStatusesFromBody = (
+  body: undefined | UpdateUtilisationReportStatusRequestBody,
+): { id: string; status: UtilisationReportReconciliationStatus }[] => {
   if (!body || typeof body !== 'object') {
-    throw new Error('Expected request body to be an object');  
+    throw new Error('Expected request body to be an object');
   }
 
   return Object.keys(body)
     .filter((key) => key.match(CHECKBOX_PATTERN.WITHOUT_GROUPS))
```

Comment:
what's the point of this without groups filter? is it just a pre filter or cna we just match against the whole checkbox pattern and if it fails to match the reportId and currentStatus we throw it away like we currently do?

Reply:
I think Christian was asking the 1st question - makes sense to me :)